mobile/src/containers/GameSessionContainer.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from "react"
+import EncryptedStorage from "react-native-encrypted-storage"
+
+const GameSessionContainer = ({ children }) => {
+    const [gameCode, setGameCode] = useState(null)
+    const [teamId, setTeamId] = useState(null)
+    const [gameSession, setGameSession] = useState(null)
+    const [teamMember, setTeamMember] = useState(null)
+
+    useEffect(() => {
+        loadLocalGameSession().then((localGameSession) => {
+            if (localGameSession) {
+                setGameSession(localGameSession.gameSession)
+                setTeamId(localGameSession.teamId)
+                setTeamMember(localGameSession.teamMember)
+                setGameCode(localGameSession.gameCode)
+            }
+        })
+    }, [])
+
+    useEffect(() => {
+        let subscription
+
+        if (gameCode) {
+            global.apiClient
+                .getGameSessionByCode(gameCode)
+                .then((gameSessionResponse) => {
+                    setGameSession(gameSessionResponse)
+                    if (gameSessionResponse?.id) {
+                        subscription =
+                            global.apiClient.subscribeUpdateGameSession(
+                                gameSessionResponse.id,
+                                (gameSessionSubscriptionResponse) => {
+                                    setGameSession(
+                                        gameSessionSubscriptionResponse
+                                    )
+                                    //TODO: update the team object everytime the game session is updated
+                                    //only update the team member if the team member is found
+                                    //if(teamMember){setTeamMember(teamMember)}
+                                }
+                            )
+                    }
+                })
+        }
+
+        return () => subscription?.unsubscribe()
+    }, [gameCode])
+
+    useEffect(() => {
+        if (gameSession) {
+            storeGameSessionLocal()
+        }
+        if (gameSession?.currentState === "CHOOSING_TRICKIEST_ANSWER") {
+            removeGameSessionLocal()
+        }
+    }, [gameSession])
+
+    async function storeGameSessionLocal() {
+        try {
+            const localGameSession = await EncryptedStorage.setItem(
+                "localGameSession",
+                JSON.stringify({
```

Comment:
Why do you need to encrypt `currentState` while that's already part of `gameSession`?